tests/test_bytes_vec_alloc.rs

```diff
@@ -1,61 +1,84 @@
 use std::alloc::{GlobalAlloc, Layout, System};
-use std::{mem, ptr};
+use std::mem;
+use std::ptr::null_mut;
+use std::sync::atomic::{AtomicPtr, AtomicUsize, Ordering};
 
 use bytes::{Buf, Bytes};
 
 #[global_allocator]
-static LEDGER: Ledger = Ledger;
+static LEDGER: Ledger = Ledger::new();
 
-struct Ledger;
+#[repr(C)]
+struct Ledger {
+    alloc_table: [(AtomicPtr<u8>, AtomicUsize); 512],
+}
 
-const USIZE_SIZE: usize = mem::size_of::<usize>();
+impl Ledger {
+    const fn new() -> Self {
+        // equivalent to size of (AtomicPtr<u8>, AtomicUsize), hopefully
+        #[cfg(target_pointer_width = "64")]
+        let tricky_bits = 0u128;
 
-unsafe impl GlobalAlloc for Ledger {
-    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
-        if layout.align() == 1 && layout.size() > 0 {
-            // Allocate extra space to stash a record of
-            // how much space there was.
-            let orig_size = layout.size();
-            let size = orig_size + USIZE_SIZE;
-            let new_layout = match Layout::from_size_align(size, 1) {
-                Ok(layout) => layout,
-                Err(_err) => return ptr::null_mut(),
-            };
-            let ptr = System.alloc(new_layout);
-            if !ptr.is_null() {
-                (ptr as *mut usize).write(orig_size);
-                let ptr = ptr.offset(USIZE_SIZE as isize);
-                ptr
-            } else {
-                ptr
+        #[cfg(target_pointer_width = "32")]
+        let tricky_bits = 0u64;
+
+        let magic_table = [tricky_bits; 512];
+
+        // i know this looks bad but all the good ways to do this are unstable or not yet
+        // supported in const contexts (even though they should be!)
+        let alloc_table = unsafe { mem::transmute(magic_table) };
```

Comment:
```suggestion
        const ELEM: (AtomicPtr<u8>, AtomicUsize) =
            (AtomicPtr::new(null_mut()), AtomicUsize::new(0));
        let alloc_table = [ELEM; 512];
```

Reply:
The problem is that they aren't Copy according to Rustc

Reply:
It appears to work correctly if you put the element in a `const`, like I did in the suggestion

Reply:
Huh

Reply:
I'll try that once this CI run finishes

Reply:
Thanks, works